lib/omniauth/login_dot_gov/authorization.rb

```diff
@@ -26,34 +27,68 @@ def redirect_url_params
           state: state,
           nonce: nonce,
         }
+
+        if aal
+          params[:aal] = aal
+        end
+
+        params
       end
 
-      def acr_values
-        if client.ial == 2
-          'http://idmanagement.gov/ns/assurance/loa/3'
+      def aal
+        if client.aal.nil?
+          nil
+        elsif VALID_AAL_VALUES.include?(client.aal.to_s)
+          client.aal
         else
-          'http://idmanagement.gov/ns/assurance/loa/1'
+          raise "Invalid AAL, choose one of #{VALID_AAL_VALUES}"
+        end
+      end
+
+      def acr_values
+        values = []
+
+        values << case client.ial
+        when 1, nil
```

Comment:
**question:** shouldn't we add validation for these values too?

Reply:
This change could also be split into a separate PR

Reply:
backed out the other changes